confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceInputFilterStream.java

```diff
@@ -305,6 +334,188 @@ private void generateBlogEvents(Map<Long, List<Long>> blogPages, Object filter,
+                    sendRight(proxyFilter, group, right, user);
+                }
+            }
+
+            if (webPreferencesStarted) {
+                proxyFilter.endWikiDocument(spaceWebPreferences, new FilterEventParameters());
+            }
+        }
+    }
+
+    private ConfluenceRightData getConfluenceRightData(ConfluenceProperties permissionProperties)
+        throws FilterException
+    {
+        String type = permissionProperties.getString(ConfluenceXMLPackage.KEY_PERMISSION_TYPE, "");
+        String groupStr = permissionProperties.getString(ConfluenceXMLPackage.KEY_SPACEPERMISSION_GROUP, null);
+        if (groupStr == null || groupStr.isEmpty()) {
+            groupStr = permissionProperties.getString(ConfluenceXMLPackage.KEY_CONTENTPERMISSION_GROUP, null);
+        }
+        String group = (groupStr == null || groupStr.isEmpty())
+                ? ""
+                : (toUserReference(getConfluenceToXWikiGroupName(groupStr)));
+
+        String user = "";
+        String userName = permissionProperties.getString(ConfluenceXMLPackage.KEY_SPACEPERMISSION_USERNAME, null);
+        if (userName == null || userName.isEmpty()) {
+            String userSubjectStr = permissionProperties.getString(ConfluenceXMLPackage.KEY_PERMISSION_USERSUBJECT, null);
+            if (userSubjectStr != null && !userSubjectStr.isEmpty()) {
+                ConfluenceProperties userProperties;
+                try {
+                    userProperties = confluencePackage.getUserImplProperties(userSubjectStr);
+                } catch (ConfigurationException e) {
+                    throw new FilterException("Failed to get user properties", e);
+                }
+
+                userName = userProperties.getString(ConfluenceXMLPackage.KEY_USER_NAME, userSubjectStr);
+            }
+        }
+
+        if (userName != null && !userName.isEmpty()) {
+            user = toUserReference(userName);
+        }
+
+        return new ConfluenceRightData(type, group, user);
+    }
+
+    private static class ConfluenceRightData
+    {
+        public final String type;
+        public final String group;
+        public final String user;
+
+        public ConfluenceRightData(String type, String group, String user) {
+            this.type = type;
+            this.group = group;
+            this.user = user;
+        }
+    }
+
+    private static void sendRight(ConfluenceFilter proxyFilter, String group, Right right, String user) throws FilterException
+    {
+        FilterEventParameters rightParameters = new FilterEventParameters();
+        // Page report object
+        rightParameters.put(WikiObjectFilter.PARAMETER_CLASS_REFERENCE, XWIKIRIGHTS_CLASSNAME);
+        proxyFilter.beginWikiObject(XWIKIRIGHTS_CLASSNAME, rightParameters);
+        proxyFilter.onWikiObjectProperty("allow", "1", FilterEventParameters.EMPTY);
+        proxyFilter.onWikiObjectProperty("groups", group == null ? "" : group, FilterEventParameters.EMPTY);
```

Comment:
There is no point passing an empty "groups", I think I would put the `if` around the #onWikiObjectProperty.